### ISSUE TITLE
Composer and PUC

### ISSUE THREAD

**Issue by regenrek** [YahnisElsts/plugin-update-checker#122]
Hi,

currently my repository has an `subfolder` named `my-plugin`
which is **required** to use the wordpress update function https://github.com/YahnisElsts/plugin-update-checker/issues/119#issuecomment-319011061

```
my-plugin
 --- my-plugin.php
 --- README.md
 --- readme.txt
 --- composer.json
 --- composer.lock
 --- vendor // here are the installed PUC files
 --- src/
 --- ...
```

Problem is that i need to place the composer.json in to the 
root directory of the repository. This is necessary to read tags from composer.json file

```
composer.json
composer.lock
 --- vendor // ?? 
my-plugin  //---- this gets fetched
 --- my-plugin.php
 --- README.md
 --- readme.txt
```

So I think there will a few problems here, since there a files in the root
directory and wordpress will throw an error with ```Update Failed: The directory structure of the update is incorrect. All files should be inside a directory named mj-acf-fields, not at the root of the ZIP archive.``` and also the `vendor` directory should be inside the my-plugin directory or it will get lost at update.

Are there any hints to make this work with composer ?

Regards

**Comment by YahnisElsts** [comment]
I think the answer is going to depend on what you're using to host the repository and package the updates. 

For example, when you download a branch or tag archive from GitHub, it automatically puts the contents of the repository in a directory named after that branch or tag (e.g.  `plugin-update-checker-4.2`). This allows you to just keep the files in the root directory - they'll still end up in a subdirectory when you actually make a release. PUC can automatically fix the directory name, so this works out fine.

Could you modify your release process to put the files in a directory?

**Comment by regenrek** [comment]
Ok, I've ended up with an far easier solution but with some drawbacks.
(For anyone who wants to use Bitbucket Server Api)

1. I'm using Bitbucket Server as main Codebase
2. Found a plugin called [Stash-Hook-Mirror](https://marketplace.atlassian.com/plugins/com.englishtown.stash-hook-mirror/server/overview) 
which can mirror Repos to any github Account.
3. Now I'm using PUC with Github.
4. Done.

Drawbacks:
5. Bitbucket Server gives me the ability to use private Repositories in our own Codebase.
This isn't possible anymore. I would prefer a Bitbucket Server standalone solution but I'm fine with that.
